packages/side-nav/src/vaadin-side-nav-item.js

```diff
@@ -0,0 +1,191 @@
+ * `suffix`  | A slot for content after the label (e.g. an icon).
+ *
+ * #### Example:
+ *
+ * ```
+ *  <vaadin-side-nav-item>
+ *     <vaadin-icon icon="vaadin:chart" slot="prefix"></vaadin-icon>
+ *     Item
+ *     <span theme="badge primary" slot="suffix">Suffix</span>
+ *   </vaadin-side-nav-item>
+ * ```
+ *
+ * @fires {CustomEvent} expanded-changed - Fired when the `expanded` property changes.
+ *
+ * @extends LitElement
+ * @mixes PolylitMixin
+ * @mixes ThemableMixin
+ * @mixes ElementMixin
+ */
+class SideNavItem extends ElementMixin(ThemableMixin(PolylitMixin(LitElement))) {
+  static get is() {
+    return 'vaadin-side-nav-item';
+  }
+
+  static get properties() {
+    return {
+      /**
+       * The path to navigate to
+       */
+      path: String,
+
+      /**
+       * Whether to show the child items or not
+       *
+       * @type {boolean}
+       */
+      expanded: {
+        type: Boolean,
+        value: false,
+        notify: true,
+        reflectToAttribute: true,
+      },
+
+      /**
+       * Whether the path of the item matches the current path.
+       * Set when the item is appended to DOM or when navigated back
+       * to the page that contains this item using the browser.
+       *
+       * @type {boolean}
+       */
+      active: {
+        type: Boolean,
+        value: false,
+        readOnly: true,
+        reflectToAttribute: true,
+      },
+    };
+  }
+
+  static get styles() {
+    return sideNavItemBaseStyles;
+  }
+
+  /** @protected */
+  get _button() {
+    return this.shadowRoot.querySelector('button');
+  }
+
+  /** @protected */
+  connectedCallback() {
+    super.connectedCallback();
+    this.setAttribute('role', 'listitem');
+    this.__updateActive();
+    this.__boundUpdateActive = this.__updateActive.bind(this);
+    window.addEventListener('popstate', this.__boundUpdateActive);
+  }
+
+  /** @protected */
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener('popstate', this.__boundUpdateActive);
+  }
+
+  /** @protected */
+  render() {
+    return html`
+      <a href="${ifDefined(this.path)}" part="item" aria-current="${this.active ? 'page' : false}">
+        <slot name="prefix"></slot>
+        <slot></slot>
+        <slot name="suffix"></slot>
+        <button
+          part="toggle-button"
+          @click="${this.__toggleExpanded}"
+          ?hidden="${!this.querySelector('[slot=children]')}"
+          aria-controls="children"
+          aria-expanded="${this.expanded}"
+          aria-label="Toggle child items"
+        ></button>
+      </a>
+      <slot name="children" role="list" part="children" id="children" ?hidden="${!this.expanded}"></slot>
+    `;
+  }
+
+  /** @private */
+  __toggleExpanded(e) {
+    if (e) {
+      e.preventDefault();
+      e.stopPropagation();
+    }
+    this.expanded = !this.expanded;
+  }
+
+  /** @private */
+  __updateActive() {
+    if (!this.path && this.path !== '') {
```

Comment:
```suggestion
    if (!this.path) {
```

Reply:
This part is to allow an empty path. Do you think it shouldn't be allowed?